native-platform/src/main/cpp/apple.cpp

```diff
@@ -71,27 +72,27 @@ Java_net_rubygrapefruit_platform_internal_jni_PosixFileSystemFunctions_listFileS
         alist.volattr = ATTR_VOL_CAPABILITIES | ATTR_VOL_INFO;
         vol_caps_buf_t buffer;
 
+        jstring mount_point = char_to_java(env, buf[i].f_mntonname, result);
+        jstring file_system_type = char_to_java(env, buf[i].f_fstypename, result);
+        jstring device_name = char_to_java(env, buf[i].f_mntfromname, result);
+        jboolean remote = (buf[i].f_flags & MNT_LOCAL) == 0;
+
         // getattrlist requires the path to the actual mount point.
         int err = getattrlist(buf[i].f_mntonname, &alist, &buffer, sizeof(buffer), 0);
         if (err != 0) {
-            mark_failed_with_errno(env, "could not determine file system attributes", result);
-            break;
-        }
-
-        if (alist.volattr & ATTR_VOL_CAPABILITIES) {
-            if ((buffer.caps.valid[VOL_CAPABILITIES_FORMAT] & VOL_CAP_FMT_CASE_SENSITIVE)) {
-                caseSensitive = (buffer.caps.capabilities[VOL_CAPABILITIES_FORMAT] & VOL_CAP_FMT_CASE_SENSITIVE) != 0;
+            env->CallVoidMethod(info, unknownFsMethod, mount_point, file_system_type, device_name, remote);
+        } else {
+            if (alist.volattr & ATTR_VOL_CAPABILITIES) {
+                if ((buffer.caps.valid[VOL_CAPABILITIES_FORMAT] & VOL_CAP_FMT_CASE_SENSITIVE)) {
+                    caseSensitive = (buffer.caps.capabilities[VOL_CAPABILITIES_FORMAT] & VOL_CAP_FMT_CASE_SENSITIVE) != 0;
```

Comment:
Nt: let's move the definition of `caseSensitive` and `casePreserving` here.

Reply:
Done.